dpctl/tensor/_usmarray.pyx

```diff
@@ -1025,10 +1025,14 @@ cdef class usm_ndarray:
         cdef c_dpmem._Memory arr_buf
         d = Device.create_device(target_device)
 
-        if (stream is None or not isinstance(stream, dpctl.SyclQueue) or
-            stream == self.sycl_queue):
+        if (stream is None or stream == self.sycl_queue):
             pass
         else:
+            if not isinstance(stream, dpctl.SyclQueue):
+                raise TypeError(
+                    "stream argument type was expected to be dpctl.SyclQueue,"
+                    f" got {type(stream)} instead"
+                )
             ev = self.sycl_queue.submit_barrier()
```

Comment:
Is there any way to write a test to verify the functionality provided by the `stream` keyword?
It seems currently the `stream` keyword is not covered in the tests.